lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,222 @@
+        new NotFound(`Object '${file}' not found while deleting in container ${container}`));
+  }
+
+  download(options) {
+    return utils.streamToPromise(this.storageClient
+        .bucket(options.container)
+        .file(options.remote)
+        .createReadStream())
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.remote}' not found`));
+  }
+
+  upload(options, buffer) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        uploadedStream.removeListener('finish', onfinish);
+        uploadedStream.removeListener('error', onerror);
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function onfinish() {
+        cleanup();
+        resolve(JSON.parse(buffer));
+      }
+      const uploadedStream = this.storageClient
+        .bucket(options.container)
+        .file(options.remote)
+        .createWriteStream();
+      uploadedStream.on('error', onerror);
+      uploadedStream.on('finish', onfinish);
+      uploadedStream.end(buffer);
+    });
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .upload({
+        container: container,
+        remote: file
+      }, new Buffer(JSON.stringify(data, null, 2), 'utf8'));
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .download({
+        container: container,
+        remote: file
+      })
+      .then((data) => JSON.parse(data))
+      .catchThrow(SyntaxError, new NotFound(`Object '${file}' not found`));
+  }
+
+  deleteSnapshot(snapshotName) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        deleteOperation.removeAllListeners();
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function oncomplete() {
+        cleanup();
+        resolve(logger.info(`Deleted snapshot ${snapshotName}`));
+      }
+
+      function onrunning() {
+        logger.info(`Deleting snapshot ${snapshotName}`);
+      }
+      var deleteOperation;
+      this.computeClient
+        .snapshot(snapshotName)
+        .delete()
+        .then(data => {
+          // Returns an Operation object that can be used to check the status of the request.
+          // All operations are event emitters. The status of each operation is polled
+          // continuously, starting only after you register a "complete" listener.
+          // https://cloud.google.com/nodejs/docs/reference/compute/0.9.x/Operation#Operation 
+          deleteOperation = data[0];
+          deleteOperation.on('complete', oncomplete);
+          deleteOperation.on('error', onerror);
+          deleteOperation.on('running', onrunning);
+        })
+        .catch(err => reject(err));
```

Comment:
can you log the error while deleting the snapshot ?

Reply:
Sure, added.